### PR TITLE
feat(Collapse): collapse add toggleAll method for issues #10818

### PR DESCRIPTION
**Pull request by LadyChatterleyLover**
### Before submitting a pull request, please make sure the following is done:

1. Read the [contributing guide](https://github.com/youzan/vant/blob/dev/.github/CONTRIBUTING.md).
2. If you've added code that should be tested, add tests.
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).

#### Title Format

type(ComponentName?)：commit message

feat(Collapse): collapse add toggleAll method for issues #10818

Example：

- docs: fix typo in quickstart
- build: optimize build speed
- fix(Button): incorrect style
- feat(Button): add color prop

Allowed Types:

- fix
- feat
- docs
- perf
- test
- types
- build
- chore
- refactor
- breaking change

### CODE REVIEW

File: packages/vant/src/checkbox/README.zh-CN.md [hunk 1]
```diff
@@ -193,6 +193,7 @@ export default {
     const checked = ref([]);
     const checkboxGroup = ref(null);
 
+
```
Comment (chenjiahan) [inline packages/vant/src/checkbox/README.zh-CN.md:196]: 多余空行建议移除

File: packages/vant/src/collapse/Collapse.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import {
   type PropType,
   type InjectionKey,
   type ExtractPropTypes,
+  ComponentPublicInstance,
```
Comment (chenjiahan) [inline packages/vant/src/collapse/Collapse.tsx:6]: 前面加个 type

File: packages/vant/src/collapse/README.md [hunk 3]
```diff
@@ -108,6 +108,52 @@ export default {
 };
 ```
 
+### Toggle All
+
+通过 `Collapse` 实例上的 `toggleAll` 方法可以实现全选与反选。
```
Comment (chenjiahan) [inline packages/vant/src/collapse/README.md:113]: 「全选与反选」—>「全部展开与全部收起」

File: packages/vant/src/collapse/README.md [hunk 4]
```diff
@@ -108,6 +108,52 @@ export default {
...
+  </van-collapse-item>
+  <van-collapse-item title="标题2" name="2">
+    技术无非就是那些开发它的人的共同灵魂。
+  </van-collapse-item>
+  <van-collapse-item title="标题3" name="3">
+    在代码阅读过程中人们说脏话的频率是衡量代码质量的唯一标准。
+  </van-collapse-item>
+</van-collapse>
+
+<van-button type="primary" @click="checkAll">全选</van-button>
```
Comment (chenjiahan) [inline packages/vant/src/collapse/README.md:128]: 「全选」—>「全部展开」

File: packages/vant/src/collapse/README.md [hunk 5]
```diff
@@ -108,6 +108,52 @@ export default {
...
+<van-button type="primary" @click="toggleAll">反选</van-button>
```
Comment (chenjiahan) [inline packages/vant/src/collapse/README.md:129]: 「反选」—>「全部收起」

File: packages/vant/src/collapse/README.md [hunk 6]
```diff
@@ -108,6 +108,52 @@ export default {
...
+
+```js
+import { ref } from 'vue';
+
+export default {
+  setup() {
+    const activeNames = ref(['1']);
+    const collapse = ref(null);
+
+    const checkAll = () => {
```
Comment (chenjiahan) [inline packages/vant/src/collapse/README.md:140]: checkAll -> openAll

File: packages/vant/src/collapse/README.md [hunk 7]
```diff
@@ -143,6 +189,37 @@ export default {
...
+// Unselect all
+collapse.toggleAll(false);
+
+// Toggle all, skip disabled
+collapse.toggleAll({
+  skipDisabled: true,
+});
+// Select all, skip disabled
+collapse.toggleAll({
+  checked: true,
```
Comment (chenjiahan) [inline packages/vant/src/collapse/README.md:218]: checked -> expanded

File: packages/vant/src/collapse/README.md [hunk 8]
```diff
@@ -143,6 +189,37 @@ export default {
...
+// Select all
+collapse.toggleAll(true);
+// Unselect all
+collapse.toggleAll(false);
+
+// Toggle all, skip disabled
+collapse.toggleAll({
+  skipDisabled: true,
+});
+// Select all, skip disabled
```
Comment (chenjiahan) [inline packages/vant/src/collapse/README.md:216]: 「Select all」-> 「Expand all」

File: packages/vant/src/collapse/README.zh-CN.md [hunk 9]
```diff
@@ -159,6 +205,37 @@ export default {
...
+// 全部取消
+collapse.toggleAll(false);
+
+// 全部反选，并跳过禁用的复选框
+collapse.toggleAll({
+  skipDisabled: true,
+});
+// 全部选中，并跳过禁用的复选框
+collapse.toggleAll({
+  checked: true,
```
Comment (chenjiahan) [inline packages/vant/src/collapse/README.zh-CN.md:234]: checked -> expanded

File: packages/vant/src/collapse/index.ts [hunk 10]
```diff
@@ -3,7 +3,7 @@ import _Collapse from './Collapse';
 
 export const Collapse = withInstall(_Collapse);
 export default Collapse;
-export type { CollapseProps } from './Collapse';
+export type { CollapseProps, CollapseToggleAllOptions } from './Collapse';
```
Comment (chenjiahan) [inline packages/vant/src/collapse/index.ts:6]: CollapseToggleAllOptions 加到文档里

**Comment by LadyChatterleyLover** [comment]
@chenjiahan 已经修改了哈，麻烦再看一下，

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
👍🏻

**Pull request merged**